src/akkudoktoreos/measurement/measurement.py

```diff
@@ -0,0 +1,262 @@
+
+        # Calculate difference in seconds
+        diff_seconds = end_datetime.diff(start_datetime).total_seconds()
+        interval_seconds = interval.total_seconds()
+
+        # Return ceiling of division to include partial intervals
+        return int(np.ceil(diff_seconds / interval_seconds))
+
+    def name_to_key(self, name: str, topic: str) -> Optional[str]:
+        """Provides measurement key for given name and topic."""
+        topic = topic.lower()
+
+        if topic not in self.topics:
+            return None
+
+        topic_keys = [key for key in self.config.config_keys if key.startswith(topic)]
+        key = None
+        if topic == "measurement_load":
+            for config_key in topic_keys:
+                if config_key.endswith("_name") and getattr(self.config, config_key) == name:
+                    key = topic + config_key[len(topic) : len(topic) + 1] + "_mr"
+                    break
+
+        if key is not None and key not in self.record_keys:
+            # Should never happen
+            error_msg = f"Key '{key}' not available."
+            logger.error(error_msg)
+            raise KeyError(error_msg)
+
+        return key
+
+    def _energy_from_meter_readings(
+        self,
+        key: str,
+        start_datetime: DateTime,
+        end_datetime: DateTime,
+        interval: Duration,
+    ) -> NDArray[Shape["*"], Any]:
+        """Calculate an  energy values array indexed by fixed time intervals from energy metering data within an optional date range.
+
+        Args:
+            key: Key for energy meter readings.
+            start_datetime (datetime): The start date for filtering the energy data (inclusive).
+            end_datetime (datetime): The end date for filtering the energy data (exclusive).
+            interval (duration): The fixed time interval.
+
+        Returns:
+            np.ndarray: A NumPy Array of the energy [kWh] per interval values calculated from
+                        the meter readings.
+        """
+        # Add one interval to end_datetime to assure we have a energy value interval for all
+        # datetimes from start_datetime (inclusive) to end_datetime (exclusive)
+        end_datetime += interval
+        size = self._interval_count(start_datetime, end_datetime, interval)
+
+        energy_mr_array = self.key_to_array(
+            key=key, start_datetime=start_datetime, end_datetime=end_datetime, interval=interval
+        )
+        if energy_mr_array.size != size:
+            logging_msg = (
+                f"'{key}' meter reading array size: {energy_mr_array.size}"
+                f" does not fit to expected size: {size}, {energy_mr_array}"
+            )
+            if energy_mr_array.size != 0:
+                logger.error(logging_msg)
+                raise ValueError(logging_msg)
+            logger.debug(logging_msg)
+            energy_array = np.zeros(size - 1)
+        elif np.any(energy_mr_array == None):
```

Comment:
would use isnan to catch numpy none and nan
        elif np.any(np.isnan(energy_mr_array)):
            # Array contains None or NaN value -> ignore
            debug_msg = f"'{key}' meter reading None or NaN: {energy_mr_array}"

Reply:
Nan are dropped by key_to_array. The sole purpose of this if is to detect if there is an empty array which is indicated by None values.